src/infrastructure/SendGridEmailClient.go

```diff
@@ -0,0 +1,48 @@
+package infrastructure
+
+import (
+	"fmt"
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+	"log"
+)
+
+type SendGridEmailClient struct {
+	apiKey      string
+	senderName  string
+	senderEmail string
+}
+
+func NewSendGridEmailClient(apiKey string, senderName string, senderEmail string) *SendGridEmailClient {
+	client := &SendGridEmailClient{apiKey: apiKey, senderName: senderName, senderEmail: senderEmail}
+
+	return client
+}
+
+func (emailClient *SendGridEmailClient) Send(recipients []string, htmlContent string) {
+	if len(recipients) == 0 {
+		return
+	}
+
+	from := mail.NewEmail(emailClient.senderName, emailClient.senderEmail)
+	firstTo := mail.NewEmail("Rate Recipient", recipients[0])
+	subject := "Current BTC to UAH rate"
+	message := mail.NewSingleEmail(from, subject, firstTo, "", htmlContent)
+
+	for i := 1; i < len(recipients); i++ {
+		personalization := mail.NewPersonalization()
+		personalization.AddTos(mail.NewEmail("Rate Recipient", recipients[i]))
+		message.AddPersonalizations(personalization)
+	}
+
+	client := sendgrid.NewSendClient(emailClient.apiKey)
+
+	response, err := client.Send(message)
+	if err != nil {
+		log.Fatalln(err)
```

Comment:
```suggestion
		return err
```
краще замість логування через log.Fatalln просто повернути помилку або залогувати через log.Println, оскільки log.Fatalln завершує виконання програми з кодом 1.

Reply:
Виправив